runtime/compiler/wasm/reader_test.go

```diff
@@ -451,7 +451,247 @@ func TestWasmReader_readTypeSection(t *testing.T) {
 	})
 }
 
-func TestWasmReader_readFunctionSection(t *testing.T) {
+func TestWASMReader_readImportSection(t *testing.T) {
+
+	t.Parallel()
+
+	read := func(data []byte) ([]*Import, error) {
+		b := buf{data: data}
+		r := WASMReader{buf: &b}
+		err := r.readImportSection()
+		if err != nil {
+			return nil, err
+		}
+		return r.Module.Imports, nil
+	}
+
+	t.Run("valid", func(t *testing.T) {
+
+		t.Parallel()
+
+		typeIDs, err := read([]byte{
+			// section size: 11 (LEB128)
+			0x8b, 0x80, 0x80, 0x80, 0x0,
+			// import count: 1
+			0x1,
+			// module length
+			0x3,
+			// module = "foo"
+			0x66, 0x6f, 0x6f,
```

Comment:
I'm not sure if we have the test case that verifies the module length match, basically we can verify that the following case should fail
```
			// module length
			0x3,
			// module = "fo", the module length, which is 2, is mismatching with the declared module length 3
			0x66, 0x6f
```

Reply:
Same as above: Parsing names is a "building block" for parsing imports (and some other parts), so I don't think we need to re-test parsing of names itself in each larger test that uses it – parsing of names is already tested by the unit test `TestWASMReader_readName`, and in particular, the subtest "invalid name" (https://github.com/onflow/cadence/blob/db04dad5bee706382ba749a7bb8f1f0f02a13305/runtime/compiler/wasm/reader_test.go#L1109) covers exactly this case: The length is larger then the length of the string that follows